client_test.go

```diff
@@ -13,3 +13,10 @@ func TestNewClientWithHTTPClientOption(t *testing.T) {
 		t.Fatalf("failed to set custom http")
 	}
 }
+
+func TestNewClientWithClientHeaders(t *testing.T) {
+	c := NewClient("api-key", WithClientHeaders(map[string]string{"foo": "bar"}))
+	if c.headers["foo"] != "bar" {
+		t.Fatalf("failed to set client headers")
+	}
+}
```

Comment:
I would add a test to see the initial headers are kept when making a request with different headers.